### PR TITLE
Remove parts of code that is not used

### PR DESCRIPTION
**Pull request by budnix**
### Context

This PR removes not used legacy code. It includes:
 * `isChildOfWebComponentTable` helper used to detect if the
   Handsontable is initialized within the custom element. The helper was 
   designed and used with Polymer and web component v0 spec. The Polymer
   project evolved into litElement (API was changed) and the v0 spec
   of the WC is deprecated;
 * DOM helpers for IE8 are removed (`pageX`, `pageY`, `stopPropagation`);
 * The event object extension has been removed from EventManager. The
   implementation was designed for web components. That implementation
   introduced the `realTarget` property that holds an element which
   causes UI action. The specified action element was extracted from custom
   element even when the component was encapsulated by shadow root;
 * The static property `isHotTableEnv` was removed;
 * From the feature helper browser check tests for IE8 ~and IE9~ was
   removed.

### How has this been tested?

Manual tests.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6525
2. https://github.com/handsontable/handsontable/issues/6024

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/helpers/dom/element.js [hunk 1]
```diff
@@ -1083,15 +1021,7 @@ export function hasHorizontalScrollbar(element) {
...
-  if (isIE8() || isIE9()) {
-    overlayElem.style.top = top;
-    overlayElem.style.left = left;
-  } else if (isSafari()) {
-    overlayElem.style['-webkit-transform'] = `translate3d(${left},${top},0)`;
-    overlayElem.style['-webkit-transform'] = `translate3d(${left},${top},0)`;
-  } else {
-    overlayElem.style.transform = `translate3d(${left},${top},0)`;
-  }
+  overlayElem.style.transform = `translate3d(${left},${top},0)`;
```
Comment (wszymanski) [inline src/helpers/dom/element.js]: It seems that `translate3d` won't work for the IE9, but we still support it.

https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function#Browser_compatibility
https://caniuse.com/#feat=transforms3d

Comment (budnix) [reply src/helpers/dom/element.js]: Yep, 👍  I went too far. I've reverted the [change](https://github.com/handsontable/handsontable/pull/6625/commits/b620dae739c195a5d9d456fa964e4303507219ff).

File: src/plugins/exportFile/test/exportFile.e2e.js [hunk 2]
```diff
@@ -94,11 +94,9 @@ describe('exportFile', () => {
 
       const result = plugin._createBlob(formatter);
 
-      if (!Handsontable.helper.isIE9()) {
```
Comment (wojciechczerniak) [inline src/plugins/exportFile/test/exportFile.e2e.js]: `isIE9` was reverted. Shouldn't this condition be reverted as well?

File: test/types/helpers/helpers.types.ts [hunk 3]
```diff
@@ -47,8 +47,6 @@ Handsontable.helper.isEdge();
 Handsontable.helper.isEmpty(1);
 Handsontable.helper.isFunction(1);
 Handsontable.helper.isIE();
-Handsontable.helper.isIE8();
-Handsontable.helper.isIE9();
```
Comment (wojciechczerniak) [inline test/types/helpers/helpers.types.ts]: `isIE9` was reverted. Should we keep the test?

Comment (budnix) [reply test/types/helpers/helpers.types.ts]: It was reverted within https://github.com/handsontable/handsontable/pull/6625/commits/b620dae739c195a5d9d456fa964e4303507219ff#diff-6a04aedfd13e4e8a5cf26a05949f4eecR50

File: src/plugins/exportFile/test/exportFile.e2e.js (hunk 2 above)
Comment (budnix) [reply src/plugins/exportFile/test/exportFile.e2e.js]: As we planned to drop support for IE9 in the next months I decided to not revert "happy test" for IE9. Moreover, I suspect that not only that test doesn't pass on IE9.

**Pull request merged**